From 2d126c1fda633b3c87bcc2278b452785e48dba0a Mon Sep 17 00:00:00 2001
From: Xiyue Deng <manphiz@gmail.com>
Date: Fri, 22 Nov 2024 14:56:16 -0800
Subject: [PATCH 1/3] Correctly construct ERT selectors S-exp (Closes:
 #1088086)

- Was under the wrong impression that ERT selectors can be specified
as the rest of the command line parameters.
  + which turns out to be the extension supported by
  `flycheck-run-tests-main'.
- Now properly make it part of the `ert-run-tests-batch-and-exit'
s-exp.
---
 dh_elpa_test | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/dh_elpa_test b/dh_elpa_test
index 82d0f77..6e0444b 100755
--- a/dh_elpa_test
+++ b/dh_elpa_test
@@ -406,14 +406,13 @@ if (@ert_files) {
     }
     if (defined $options->{_}->{'ert_helper'}) {
         push @args, ("-l", $options->{_}->{'ert_helper'});
+    } elsif (defined $options->{_}->{'ert_selectors'}) {
+        my $ert_selectors = $options->{_}->{'ert_selectors'};
+        push @args, ("--eval", "(ert-run-tests-batch-and-exit ${ert_selectors})");
     } else {
         push @args, ("--eval", "(ert-run-tests-batch-and-exit)");
     }
 
-    if (defined $options->{_}->{'ert_selectors'}) {
-        push @args, $options->{_}->{'ert_selectors'};
-    }
-
     print_and_doit(@args);
 }
 
-- 
2.45.2

